lib/src/baseFormValidation.ts

```diff
@@ -40,8 +60,13 @@ export class BaseFormValidation implements FormValidation {
     });
   }
 
-  validateField(vm: any, key: string, value: any, filter?: any): Promise<FieldValidationResult> {
-    return this.validationEngine.validateSingleField(vm, key, value, filter);
+  private addFieldValidation(key: string, validationFunction: FieldValidationFunction, trigger?: ValidationFilter): FormValidation {
+    this.validationEngine.addFieldValidation(key, validationFunction, trigger);
+    return this;
+  }
+
+  validateField(vm: any, key: string, value: any, filter?: ValidationFilter): Promise<FieldValidationResult> {
```

Comment:
We need to be consistent when naming `trigger` or `filter`. Select one of this and use it always.